### PR TITLE
[Bugfix:TAGrading] Fix Progress Bar for Peer Graders

### PR DESCRIPTION
**Pull request by NiaHeermance**
### What is the current behavior?
When logged in as a peer grader, progress bar displays 100% even when one has not graded a single component.

### What is the new behavior?
The progress bar now accurately displays the number of components you have left to grade.

### To Test
1. As the instructor, create a gradeable with peer grading components.
2. Login as at least three students and submit to this gradeable.
3. As the instructor, in the gradeable's configuration, generate peer grading assignments in the peer grading tab.
4. As one of the students who submitted, grade one component of another student. Refresh the page and notice the progress bar updates.

Solves issue #10050 .

### CODE REVIEW

**Comment by VVVITH** [comment]
Hi, I have reviewed from the instructor view, looks all right, but haven't test from the peer grader view. Which account should I use to test peer grader view? I've tried many student account but neither of them have the peer grader access.
Also, the code on new line 1762, is there a problem and waiting to be fixed?

**Comment by NiaHeermance** [comment]
Hello, I've just updated the main post with instructions how to test, so you can see the peer view with that. As for the comment, I do need to address it, as I have now learned there is peer grading for team gradeables.

**Review by VVVITH — COMMENTED** [review]
As the test instruction, I did test the case for 0 percentage of peer grade. It does change from 100 to 0. But when I try to do peer grade to see more progress bar cases. I meet something weird. Login with the peer grader student, I can access the grading and the actual grade page, but everytime I want to make change on the grading component and save my grade, it calls an error. Error message attached.
截屏2024-01-30 02 02 37
I'm not clear what settings I have to meet this situtation.

**Comment by williamlin6803** [comment]
Hi, I have reviewed this and it looks good to me. One critique I have is that: For example, some variables could benefit from more descriptive names. For example, $total_submitted could be renamed to something like $total_submitted_peer to make it clear that it's specifically related to peer grading.

**Comment by ZheyuDeng** [comment]
Tested the code and follow test steps provided looks good!

**Review by thegame615 — COMMENTED** [review]
I followed the steps and was able to replicate the issue on main and verify that the issue was fixed on the branch. I imagine that we will probably have many issues associated with peer grading in the future.

**Review by ZheyuDeng — COMMENTED** [review]
After the php lint is fixed should be good to go

**Review by ArnavPawar — APPROVED** [review]
Has a PHP fix required but this seems to fix the issue with what is currently in the main branch. I don't get the same issue when testing it in your branch. As for the code it looks good seemed to have just added team checks and change the grading form if it was a team assignment.

**Review by JWei17 — APPROVED** [review]
This looks good to me, at less I didn't find out any bugs.

**Review by arougey — APPROVED** [review]
Code works as expected and is understandable

File: site/app/controllers/grading/ElectronicGraderController.php [hunk 1]
```diff
@@ -1917,6 +1926,31 @@ public function showGrading(
         $this->core->getOutput()->renderOutput(['grading', 'ElectronicGrader'], 'popupSettings');
     }
 
+    /**
+     * Given a list of peers to grade, figure out how many teams we are
+     * actually grading.
+     *
+     * @param array<string> $people_assigned People we are being grading
+     * @param array<Team> $teams All teams for this gradeable.
+     * @return The number of teams this peer is grading.
```
Comment (williamjallen) [inline site/app/controllers/grading/ElectronicGraderController.php:1935]: The "The" after the `@return` PHPDoc annotation is being interpreted as a type by PHPStan, and is causing a PHPStan error.

File: site/app/controllers/grading/ElectronicGraderController.php [hunk 2]
```diff
@@ -1917,6 +1926,31 @@ public function showGrading(
         $this->core->getOutput()->renderOutput(['grading', 'ElectronicGrader'], 'popupSettings');
     }
 
+    /**
+     * Given a list of peers to grade, figure out how many teams we are
+     * actually grading.
+     *
+     * @param array<string> $people_assigned People we are being grading
+     * @param array<Team> $teams All teams for this gradeable.
```
Comment (williamjallen) [inline site/app/controllers/grading/ElectronicGraderController.php:1934]: You need to add the right namespace for `Team` to fix the PHPStan error.

**Review by williamjallen — CHANGES_REQUESTED** [review]
Everything looks decent, but PHPStan is still failing.  This should be on track to merge once the PHPStan errors are resolved.

**Comment by thegame615** [comment]
Relevant issue is closed by [#10272](https://github.com/Submitty/Submitty/pull/10272)

**Pull request closed without merging**